app/post.go

```diff
@@ -373,6 +387,36 @@ func (a *App) CreatePost(c request.CTX, post *model.Post, channel *model.Channel
 	return rpost, nil
 }
 
+func (a *App) validateVoiceMessage(post *model.Post) *model.AppError {
+	fileIds := post.FileIds
+	nbFiles := len(fileIds)
+	if nbFiles == 0 {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.no_file.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+	}
+
+	if nbFiles > 1 {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.too_many_files.app_error", nil, "id="+post.Id, http.StatusBadRequest)
+	}
+
+	fileInfo, err := a.GetFileInfo(fileIds[0])
+	if err != nil {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.unable_get_file_info.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusInternalServerError).Wrap(err)
+	}
+	if fileInfo == nil {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.file_not_found.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusBadRequest)
+	}
+
+	if !fileInfo.CanBeUsedForVoiceMessage() {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.invalid_file_type.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusBadRequest)
+	}
+
+	if fileInfo.Size > *a.Config().FileSettings.MaxVoiceMessagesFileSize {
+		return model.NewAppError("validateVoicePost", "api.post.validate_voice_post.file_too_big.app_error", nil, "id="+post.Id+" fileId="+fileIds[0], http.StatusBadRequest)
+	}
```

Comment:
We discussed this a bit and I think the idea is to transition to have a max duration rather than file size which makes more sense in this context. The max file size would still be enforced by the global setting on file uploads.

Reply:
Yes, I do remember! I didn't want to apply the change before our demo as I don't know if the frontend/mobile was relying on this or not, and switching to a time limit will definitely call for some changes, like not letting the user record for 2minutes and deny his message when he's sending it.

Reply:
Of course, we need proper UI in place based on the setting. The idea is that the recording will force stop upon reaching the max duration and likely an explanatory message will be shown to the user. /cc @M-ZubairAhmed @larkox

Reply:
Yes we need the design of that, i added 2 min auto shut off for recording, but commented it out. cc @matthewbirtch

Reply:
@matthewbirtch Indeed, that's what I had in mind to at least inform the user of what's happening. Of course we could implement more advanced ways to warn the user prior to recording stopping but probably out of scope for now. @esethna and @Rishselva please feel free chime in.

Reply:
> the design need here is a way to display a message that the recording stopped because the user reached the maximum duration?

Makes sense to me!

Reply:
Would something like this work? @esethna @streamer45 @M-ZubairAhmed @JulienTant 

![image](https://user-images.githubusercontent.com/2040554/197216582-3a94d215-59a7-4f3e-b956-659027141a9c.png)

Reply:
+1 from me

Reply:
@matthewbirtch this seems to be slightly different from the existing error messages we have for file attachments, Since we will be hooking into the same error system.
<img width="719" alt="image" src="https://user-images.githubusercontent.com/17708702/197242409-079b9167-f5f0-40a5-99db-25f146970304.png">

Reply:
@M-ZubairAhmed I didn't want to display it as an error because it's not really an error from a user's standpoint. The file still saved.